crates/sui-graphql-rpc/src/types/event.rs

```diff
@@ -56,13 +43,52 @@ pub(crate) struct EventFilter {
     // pub not
 }
 
-#[ComplexObject]
+#[Object]
 impl Event {
     /// The Move module that the event was emitted in.
     async fn sending_module(&self, ctx: &Context<'_>) -> Result<Option<MoveModule>> {
+        let sending_package = SuiAddress::from_bytes(&self.stored.package)
+            .map_err(|e| Error::Internal(e.to_string()))
+            .extend()?;
         ctx.data_unchecked::<PgManager>()
-            .fetch_move_module(self.sending_package, &self.sending_module)
+            .fetch_move_module(sending_package, &self.stored.module)
             .await
             .extend()
     }
+
+    /// Addresses of the senders of the event
+    async fn senders(&self) -> Option<Vec<Address>> {
+        let result: Option<Vec<Address>> = self
+            .stored
+            .senders
+            .iter()
+            .filter_map(|sender| {
+                sender.as_ref().map(|sender| {
+                    SuiAddress::from_bytes(sender)
+                        .map(|sui_address| Address {
+                            address: sui_address,
+                        })
+                        .map_err(|e| eprintln!("Unexpected None value in senders array: {}", e))
+                        .ok()
+                })
+            })
+            .collect();
+
+        result
+    }
+
```

Comment:
We shouldn't use `eprintln!` here.  Recommend returning an internal error instead.

```suggestion
    async fn senders(&self) -> Result<Option<Vec<Address>>> {
        let mut addrs = Vec::with_capacity(self.stored.senders.len());
        for sender in &senders {
            let Some(sender) = &sender else { continue };
            let address = SuiAddress::from_bytes(sender)
                .map_err(|e| Error::Internal(format!("Failed to deserialize address: {e}")))
                .extend()?;
            addrs.push(Address { address });
        }
        Ok(Some(addrs))
    }

```

Reply:
Thanks this is much cleaner